libs/header/sync/sync_getter.go

```diff
@@ -0,0 +1,31 @@
+package sync
+
+import (
+	"context"
+	"sync"
+
+	"github.com/celestiaorg/celestia-node/libs/header"
+)
+
+// syncGetter is a Getter wrapper that ensure only one Head call happens at the time
+type syncGetter[H header.Header] struct {
```

Comment:
I would call it `syncHeadGetter` if you are expecting a single `Head` method. But I do not insist. Just a nit.

Reply:
I didn't call it so because we still use it for other purposes besides Head. syncGetter just overrides one method